spec/01-unit/01-conf/01-conf_loader_spec.lua

```diff
@@ -172,29 +191,55 @@ describe("Configuration loader", function()
       assert.equal("no file at: inexistent", err)
       assert.is_nil(conf)
     end)
-    it("requires cert and key if SSL is enabled", function()
+    it("returns a DNS error when both a resolver and dnsmasq are enabled", function()
       local conf, err = conf_loader(nil, {
-        ssl = true
+        dnsmasq = true,
+        dns_resolver = "8.8.8.8:53"
       })
-      assert.equal("ssl_cert required if SSL enabled", err)
+      assert.equal("when specifying a custom DNS resolver you must turn off dnsmasq", err)
       assert.is_nil(conf)
-
+      
       conf, err = conf_loader(nil, {
-        ssl = true,
+        dnsmasq = false,
+        dns_resolver = "8.8.8.8:53"
+      })
+      assert.is_nil(err)
+      assert.truthy(conf)
+    end)
+    it("cluster_ttl_on_failure cannot be lower than 60 seconds", function()
+      local conf, err = conf_loader(nil, {
+        cluster_ttl_on_failure = 40
+      })
+      assert.equal("cluster_ttl_on_failure must be at least 60 seconds", err)
+      assert.is_nil(conf)
+    end)
+    it("requires both SSL cert and key", function()
+      local conf, err = conf_loader(nil, {
         ssl_cert = "/path/cert.pem"
       })
-      assert.equal("ssl_cert_key required if SSL enabled", err)
+      assert.equal("ssl_cert_key must be enabled", err)
       assert.is_nil(conf)
+
+      conf, err = conf_loader(nil, {
+        ssl_cert_key = "/path/key.pem"
+      })
+      assert.equal("ssl_cert must be enabled", err)
+      assert.is_nil(conf)
+
+      conf, err = conf_loader(nil, {
+        ssl_cert = "/path/cert.pem",
+        ssl_cert_key = "/path/key.pem"
+      })
+      assert.falsy(err)
+      assert.truthy(conf)
     end)
     it("returns all errors in ret value #3", function()
       local conf, _, errors = conf_loader(nil, {
-        ssl = true,
         cassandra_repl_strategy = "foo"
       })
-      assert.equal(2, #errors)
+      assert.equal(1, #errors)
```

Comment:
This test specifically tests that the third return value can return more than a single error, it should still do this, potentially with different input if `ssl = true` now gives another result.